walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -233,4 +240,44 @@
             MoreExecutors.directExecutor()
         );
     }
+
+    /**
+     * Gets all the applicant's interested jobs given the params.
+     *
+     * @param applicantId The applicant's userId.
+     * @param pageSize The the number of jobs to be shown on the page.
+     * @param pageIndex The page number on which we are at.
+     * @return Future of the JobPage object.
+     */
+    public Future<JobPage> fetchInterestedJobPage(String applicantId, int pageSize, int pageIndex) throws IOException {
+        CollectionReference applicantAccountsCollection = FireStoreUtils.getFireStore().collection(APPLICANT_ACCOUNTS_COLLECTION);
+
+        DocumentReference docRef = applicantAccountsCollection.document(applicantId);
+
+        return ApiFutures.transform(
+            docRef.get(),
+            documentSnapshot -> {
+                ImmutableList.Builder<Job> jobListBuilder = ImmutableList.builder();
+                List<String> interestedList = (List<String>) documentSnapshot.get("interestedJobs");
+
+                for (String jobId: interestedList) {
+                    try {
+                        Optional<Job> job = fetchJob(jobId).get(5, TimeUnit.SECONDS);
+                        if (job.isPresent()) {
```

Comment:
job must be present AND not in DELETED state. You also have to decide what to do with jobs that are in non-published state (meaning not publicly available anymore) here you are returning them so the client-side will have to filter them out OR gray them out somehow on the screen.

Reply:
I added a check to make sure the job is in ACTIVE state